dace/frontend/python/newast.py

```diff
@@ -1561,6 +1580,9 @@ def parse_program(self, program: ast.FunctionDef, is_tasklet: bool = False):
             if vname.startswith('__return'):
                 if isinstance(self.sdfg.arrays[arrname], data.View):
                     # In case of a view, make a copy
+                    # NOTE: If we are at the top level SDFG (not always clear),
+                    # and it is a View of an input array, can we return a NumPy
+                    # View directly?
```

Comment:
probably not, a non-transient array acts as a view

Reply:
I am considering two cases here that might be interesting to users (and probably difficult to implement properly, if possible at all):

The first case is returning a View of an input. For example:
```python
def foo(x):
    return x[1:5]
    
A = np.arange(10)
B = foo(A)
assert (B.base is A)
```

The second case is returning a View of transient data:
```python
def foo():
    x = np.arange(10)
    return x[1:5]

B = foo()
assert (B.base is not B)
```

Any work on such cases will have to be done on another PR. However, do you think that there would be any benefit in trying to support such codes or is it better to discourage the users from such coding practices?

Reply:
No, we do want that behavior possible and we can assume that a copy will be made, as long as it's documented